### PR TITLE
fix: don't check nonce in calls

### PR DESCRIPTION
**Pull request by Wodann**
[This change](https://github.com/bluealloy/revm/commit/b25b44aac8132bd7086faf3bc3e5f79cbf0ded49) was required to our fork of `revm`. We'll need to port this to the latest version of `revm`

### CODE REVIEW

File: crates/edr_evm/src/runtime.rs
Comment (Wodann) [inline crates/edr_evm/src/runtime.rs:69]: This disables nonce checks for all calls; i.e `eth_call`, `debug_traceCall`, and `eth_estimateGas`. Is this desired @fvictorio ?

**Review by agostbiro — APPROVED** [review]
Thanks, LGTM, but I'd wait for @fvictorio 's review as well before merging.

File: crates/edr_evm/src/runtime.rs
Comment (fvictorio) [reply crates/edr_evm/src/runtime.rs:69]: I think it's desired, yes. I also checked and this matches geth's behavior.

File: crates/edr_evm/src/mempool.rs
Comment (fvictorio) [inline crates/edr_evm/src/mempool.rs:340]: I'm curious why this was needed. I would've guessed that having an entry in the pending/future transactions map whose value is an empty list wouldn't be a problem.

**Review by fvictorio — APPROVED** [review]

**Pull request merged**
